lib/spack/spack/compilers/__init__.py

```diff
@@ -413,6 +413,14 @@ def get_compilers(config, cspec=None, arch_spec=None):
             assert arch_spec is None
 
         if arch_spec and target and (target != family and target != 'any'):
+            # If the family of the target is the family we are seeking,
+            # there's an error in the underlying configuration
+            if llnl.util.cpu.targets[target].family == family:
+                msg = ('the "target" field in compilers.yaml accepts only '
+                       'target families [substitute "{0}" with "{1}"'
```

Comment:
This should say `substitutue "{1}" for "{0}"` or `replace "{0}" with "{1}"`. (although in the first case we would probably switch the order of the arguments so they are in order still).